src/frontend/gatsby-node.js

```diff
@@ -1,4 +1,56 @@
 const path = require(`path`);
+const fs = require('fs').promises;
+const fetch = require('node-fetch');
+
+async function downloadPhoto(url, filename) {
+  const res = await fetch(url);
+  const buf = await res.buffer();
+  await fs.writeFile(filename, buf);
+}
+
+async function shouldDownload(filename) {
+  try {
+    await fs.access(filename, fs.FS_OK);
+    // File already exists, skip
+    return false;
+  } catch (err) {
+    // No such file, we need it
+    return true;
+  }
+}
+
+exports.onPreInit = async function () {
+  const clientId = process.env.unsplashClientId || '';
```

Comment:
What happens if you don't have a `clientId`?  Will it still do something, or should we bail right here if it's not set?

Also, can you change the names of these `process.env.UNSPLASH_CLIENT_ID` and `process.env.COLLECTION_ID`, and add it to `env.example` with comments above on what it is, and what kind of value it takes?

Reply:
Will make the change, and the one image is due to it being the "sample" image which ensures everyone has a local image available. I could add the bail out logic early, makes sense that if they don't have clientID to jump out.